core/include/debug.h

```diff
@@ -31,6 +31,7 @@
 
 #include <stdio.h>
 #include "sched.h"
+#include "thread.h"
```

Comment:
That's a bug in the user's file then.. I'm against putting includes in `#if` guards.